File: euler/P1.py
# Multiples of 3 and 5
# Problem 1

if __name__ == '__main__':
    # Generates all numbers from 0 to 999 that are a multiple of
    # 3 or a multiple of 5 (inclusively), and prints the sum
    print sum(i for i in range(1000) if not (i % 3 and i % 5))

File: euler/c++/P1.cc
/*
  Multiples of 3 and 5
  Problem 1
*/

#include <iostream>

const uint64_t UPPER_LIMIT = 1000;

int main(int argc, char *argv[]) {
  uint64_t sum = 0;
  for(uint64_t i = 0; i < UPPER_LIMIT; i++) {
    sum += i * !(i % 3 && i % 5);
  }
  std::cout << sum << std::endl;
  return 0;
}
